### PR TITLE
Workaround missing "Optional 2" display

### PR DESCRIPTION
**Pull request by koppor**
In biblatex mode, when resetting the preferences, JabRef displayed "Optional fields" and "Optional fields 2":

312461887-4dc364ff-7cd1-4b8f-94ac-1fe7ef0cc424

When opening the preferences, navigating to "customize entry types" and immediately saving, the "Optional fields 2" disappears and only "Optional fields" are displayed. Moreover, there is no choice between required fields (e.g., author or editor), JabRef now requires both.

This PR adds a work around: Customized entry types are only stored by JabRef if there was any change. If the fields required and optional are the same as in the respective standard type, there is no writing (and no handling). Thus  "Optional fields 2" will stay for non-customized types.

- Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/378
- Follow-up to https://github.com/JabRef/jabref/pull/5799

### Mandatory checks

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/test/java/org/jabref/model/entry/BibEntryTypesManagerTest.java [hunk 1]
```diff
@@ -57,7 +57,7 @@ void setUp() {
     }
 
     private BibEntryType getStandardArticleType(BibDatabaseMode mode) {
-        return entryTypesManager.getEntryTypes(mode).standardTypes.stream().filter(t -> StandardEntryType.Article.equals(t.getType())).findAny().get();
+        return entryTypesManager.getEntryTypes(mode).standardTypes.stream().filter(t -> StandardEntryType.Article == t.getType()).findAny().get();
```
[the remark this answers is not in the document]
Comment (koppor) [reply src/test/java/org/jabref/model/entry/BibEntryTypesManagerTest.java:60]: There is only one standard type? Thus, I thought, `findAny` could be faster?

[the remark this answers is not in the document]
Comment (calixtus) [reply src/test/java/org/jabref/model/entry/BibEntryTypesManagerTest.java:60]: does it matter in a test? have we ever had problems with reliability of these tests?

[the remark this answers is not in the document]
Comment (koppor) [reply src/test/java/org/jabref/model/entry/BibEntryTypesManagerTest.java:60]: I think, somewhere in the main code, I also had it. Chris might be afraid of "Heisenbugs" :)

**Comment by koppor** [comment]
I added a test to illustrate the issue - [`e497592` (#11022)](https://github.com/JabRef/jabref/pull/11022/commits/e497592dca478ea05b7c5e1554e1816488506c96)

**Comment by Siedlerchr** [comment]
I am now constantly getting library changed messages when entering texst:
[otherkey.bib.zip](https://github.com/JabRef/jabref/files/14641733/otherkey.bib.zip)

**Comment by koppor** [comment]
Cannot reproduce on `main`. Happens in this branch.

Could reproduce once in this branch.

Restarted JabRef startet with the file again. Could not trigger the issue again. -- @Siedlerchr Do you remember what steps you did? :)

**Comment by koppor** [comment]
I could reproduce it when adding spaces to 	the citation key and then pressing "Dismiss Changes". Then saving. Then "Dismiss Changes".

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**